File: frontend/src/shared/types/agent.ts
export type AgentCitation = {
  title: string;
  source_type: string;
  snippet: string;
};

export type ChatMessage = {
  role: "user" | "assistant" | string;
  content: string;
};

export type ChatResponse = {
  session_id: string;
  answer: string;
  citations: AgentCitation[];
  context_summary: string;
  suggested_next_questions: string[];
  ai_generated: boolean;
};

File: backend/app/schemas/agent.py
from pydantic import BaseModel, Field


class ChatMessage(BaseModel):
    role: str
    content: str


class ChatRequest(BaseModel):
    message: str = Field(min_length=1)
    scenario: str = "general"
    session_id: str = "demo_session"
    history: list[ChatMessage] = Field(default_factory=list)


class Citation(BaseModel):
    title: str
    source_type: str
    snippet: str


class ChatResponse(BaseModel):
    session_id: str
    answer: str
    citations: list[Citation] = Field(default_factory=list)
    context_summary: str
    suggested_next_questions: list[str] = Field(default_factory=list)
    ai_generated: bool = True
